engine/src/main/java/org/hibernate/search/engine/metadata/impl/BridgeDefinedField.java

```diff
@@ -57,4 +67,12 @@ public FieldType getType() {
 	public String toString() {
 		return "BridgeDefinedField [name=" + absoluteName + ", type=" + type + "]";
 	}
+
+	public void add(Class class1, Object elasticsearch) {
+		extra.put( class1, elasticsearch );
+	}
+
+	public <T> T mappedOn( Class<T> class1 ) {
+		return (T) extra.get( class1 );
```

Comment:
Unless I'm mistaken, there should probably be a `@SuppressWarnings("unchecked")` on this method, to avoid a warning. I know there already are tons of them, but we could at least refrain from adding new ones...

Another (maybe better) solution would be to use the class to do the cast:

```
  return class1.cast( extra.get( class1 ) );
```